libgambatte/include/fptrs.h

```diff
@@ -0,0 +1,21 @@
+#ifndef FPTRS_H
+#define FPTRS_H
+
+#include <stddef.h> /* size_t */
+
+#ifdef __cplusplus
+namespace gambatte {
+extern "C" {
```

Comment:
extern "C" here is a bit interesting, *technically* this is correct, although in practice it typically doesn't actually matter, as most (if not all practical) C++ implementations wouldn't have any issues with a C++ linkage function being assigned to a C linkage function pointer and vice versa (as they function the same at runtime), although this isn't actually guaranteed by the standard (C++ linkage could entail a different calling convention compared to the same function being assigned C linkage, in practice this never occurs).

On that, technically this is not correctly handled in gambatte.h as they all would use C++ (default) linkage, so if we want that to be ""correctly"" done, all those callback definitions need extern "C".

Reply:
On the off chance that there is a compiler out there that defaults to using a different calling convention between C and C++, I went ahead and declared the other function prototypes as `extern "C"` for consistency.

Moreover, to make sure the C API is actually usable from C, I added a simple test in [c75fe83](https://github.com/pokemon-speedrunning/gambatte-core/pull/19/commits/c75fe837cbcbe7b857fd74783525538a287e8cc0) that creates and deletes a GB object using the C API.